Multi-tenancy bundle

takeit opened the pull request:
See the README.md file for the documentation.

takeit committed:
initialize multitenancy bundle

fix composer.json typo

fixes

add type casts

added new line

update readme

added sensiolabs insights badge

improvements

ahilles107 commented on .travis.yml:
```diff
@@ -0,0 +1,35 @@
+  directories:
+    - $HOME/.composer/cache/files
+
+matrix:
```
don't build matrix, just use env like here: https://github.com/willdurand/Hateoas/blob/master/.travis.yml#L17

takeit replied:
but whats the difference then ? the goal will be the same

ahilles107 replied:
File is smaller, and travis set's env before running job. I had travis error (that he can't set php env) on some jobs with this (current) way.

And at end you have allways testsed only --prefer-stable (as you have two flags at once). Just check how travis build looks - https://travis-ci.org/SuperdeskWebPublisher/SWPMultiTenancyBundle/builds/105408330. It's all wrong. 

We should tests every php with --prefer-lowest and --prefer-stable (jobs without flags are using allways --prefer-stable as defaults - as we don't have lock file here).

ahilles107 commented on Initializer/PHPCRBasePathsInitializer.php:
```diff
@@ -0,0 +1,137 @@
+        return $this->genereteBasePaths($tenants);
+    }
+
+    private function genereteBasePaths(array $tenants = array())
```
if it's private and you don't re-use it on this same class why not keep code in getBasePaths function (when you have just one line extra).

takeit replied:
because its more readable, it doesnt matter there is a second private method or third one. The code is more clean because each method does a different thing. (getting tenants, generating base path and at the end creating them). Isnt it better to read it like that than reading it having all in one method ?

ahilles107 replied:
BTW. it should be `generateBasePaths` not `genereteBasePaths` (e => a). 

But as only thing what getBasePaths is doing is calling genereteBasePaths then one of them is not needed. If you would keep somewhere those basepaths inside class then getBasePaths would make sense. Now you always generate them. One of them should be removed. It's will just make stack trace longer in case of exceptions.

takeit committed:
added doctrine extensions and fixes

takeit committed:
bundle improvements

bump lib version

increase symfony version to 2.4 to make use of requeststack

fix scrutinizer issues

takeit committed:
added doctrine extensions

takeit committed:
provide default configuration

takeit committed:
update readme file

fix typo

takeit committed:
config fixes, initializer fix

takeit committed:
added command to create tenant

takeit committed:
added create tenant command tests

takeit committed:
added class representing page

added missing param

takeit committed:
improve console command

improve create tenant console command

takeit committed:
update composer.json

ahilles107 commented:
:+1:
